sdl_viewer/src/main.rs

```diff
@@ -260,38 +264,40 @@ impl NodeViewContainer {
         while let Ok((node_id, node_data)) = self.node_data_receiver.try_recv() {
             // Put loaded node into hash map.
             self.requested.remove(&node_id);
-            self.node_views
-                .insert(node_id, NodeView::new(program, node_data));
-            // TODO(sirver): Use a LRU Cache to throw nodes out that we haven't used in a while.
+            self.cache.insert(node_id, NodeView::new(program, node_data));
         }
 
-        match self.node_views.entry(*node_id) {
-            Entry::Vacant(_) => {
-                // Limit the number of requested nodes because after a camera move
-                // requested nodes might not be in the frustum anymore.
-                if !self.requested.contains(&node_id) && self.requested.len() < 10 {  
-                    self.requested.insert(*node_id);
-                    self.node_id_sender.send(*node_id).unwrap();
-                }
-                None
+        if !self.cache.contains_key(node_id) {
```

Comment:
Restructure to avoid right drift: if `self.cache.contains_key() { return ... };` then do the other case without else clause.